com.unity.multiplayer.mlapi/Tests/Runtime/NetworkObject/NetworkObjectOnSpawnTests.cs

```diff
@@ -0,0 +1,233 @@
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI;
+using MLAPI.RuntimeTests;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace MLAPI.RuntimeTests
+{
+    public class NetworkObjectOnSpawnTests
+    {
+        private GameObject m_Prefab;
+
+        private int m_OriginalTargetFrameRate;
+
+        /// <summary>
+        /// Tests that instantiating a <see cref="NetworkObject"/> and destroying without spawning it
+        /// does not run <see cref="NetworkBehaviour.OnNetworkSpawn"/> or <see cref="NetworkBehaviour.OnNetworkSpawn"/>.
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator InstantiateDestroySpawnNotCalled()
+        {
+            var gameObject = new GameObject("InstantiateDestroySpawnNotCalled_Object");
+            var networkObject = gameObject.AddComponent<NetworkObject>();
+            var fail = gameObject.AddComponent<FailWhenSpawned>();
+
+            yield return null;
+
+            // instantiate
+            var instance = Object.Instantiate(gameObject);
+            yield return null;
+
+            // destroy
+            Object.Destroy(instance);
+            yield return null;
+        }
+
+        private class FailWhenSpawned : NetworkBehaviour
+        {
+            public override void OnNetworkSpawn()
+            {
+                Assert.Fail("Spawn should not be called on not spawned object");
+            }
+
+            public override void OnNetworkDespawn()
+            {
+                Assert.Fail("Depawn should not be called on not spawned object");
+            }
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            // Just always track the current target frame rate (will be re-applied upon TearDown)
+            m_OriginalTargetFrameRate = Application.targetFrameRate;
+
+            // Since we use frame count as a metric, we need to assure it runs at a "common update rate"
+            // between platforms (i.e. Ubuntu seems to run at much higher FPS when set to -1)
+            if (Application.targetFrameRate < 0 || Application.targetFrameRate > 120)
+            {
+                Application.targetFrameRate = 120;
+            }
```

Comment:
not this PR but I think we should bake this into `MultiInstanceHelper` itself instead of bumping into issues all the time and fixing this ad-hoc here and there (I know Noel, myself and now yourself bumped into this — in fact, I spend a few hours in one of PRs to boils it down to this issue, wasn't a great experience at all)
just noting here, will put up a PR for this very soon.

Reply:
will no longer be needed with #893